site/src/pages/WorkspacesPage/WorkspaceSearch/TemplateMenu.tsx

```diff
@@ -0,0 +1,99 @@
+import MenuItem from "@mui/material/MenuItem";
+import MenuList from "@mui/material/MenuList";
+import { useState } from "react";
+import { useQuery } from "react-query";
+import { templates } from "api/queries/templates";
+import { Loader } from "components/Loader/Loader";
+import { MenuButton } from "components/Menu/MenuButton";
+import { MenuCheck } from "components/Menu/MenuCheck";
+import { MenuNoResults } from "components/Menu/MenuNoResults";
+import { MenuSearch } from "components/Menu/MenuSearch";
+import {
+  PopoverContent,
+  PopoverTrigger,
+  usePopover,
+  withPopover,
+} from "components/Popover/Popover";
+import { TemplateAvatar } from "components/TemplateAvatar/TemplateAvatar";
+
+type TemplateMenuProps = {
+  organizationId: string;
+  selected: string | undefined;
+  onSelect: (value: string) => void;
+};
+
+export const TemplateMenu = withPopover<TemplateMenuProps>((props) => {
+  const popover = usePopover();
+  const { organizationId, selected, onSelect } = props;
+  const [filter, setFilter] = useState("");
+  const templateOptionsQuery = useQuery({
+    ...templates(organizationId),
+    enabled: selected !== undefined || popover.isOpen,
+  });
+  const options = templateOptionsQuery.data
+    ?.filter((t) => {
+      const f = filter.toLowerCase();
+      return (
+        t.name?.toLowerCase().includes(f) ||
+        t.display_name.toLowerCase().includes(f)
+      );
+    })
+    .map((t) => ({
+      label: t.display_name ?? t.name,
+      value: t.id,
+      avatar: <TemplateAvatar size="xs" template={t} />,
+    }));
```

Comment:
I feel like this would be a good candidate for [React Query's `select` property](https://tanstack.com/query/v4/docs/framework/react/reference/useQuery). Main benefits:

1. The `select` callback is set up so that the data is always defined. No need to worry about null checks
2. Transforming the data via `select` doesn't touch the underlying query data. The transformed data is never put in the query cache
3. The transformation logic only re-runs when the underlying data changes. Right now, `options` is re-evaluated on every render

Reply:
Another good idea!